pkg/daemon/daemon.go

```diff
@@ -628,7 +595,7 @@ func (dn *Daemon) nodeStateSyncHandler() error {
 			}
 		}
 
-		if dn.openshiftContext.IsOpenshiftCluster() && !dn.openshiftContext.IsHypershift() {
+		if dn.platformHelpers.IsOpenshiftCluster() && !dn.platformHelpers.IsHypershift() {
```

Comment:
maybe platformHelpers interface should have a : GetClusterType() instead of using vars.ClusterType

Reply:
going back to our discussion about globals. lets keep this one as is.